runtime/lua/vim/lsp.lua

```diff
@@ -502,16 +514,8 @@ function lsp.start_client(config)
         or error(string.format("not found: %q request callback for client %q.", method, client.name))
     end
     local _ = log.debug() and log.debug(log_prefix, "client.request", client_id, method, params, callback, bufnr)
-    -- TODO keep these checks or just let it go anyway?
-    if (not client.resolved_capabilities.hover and method == 'textDocument/hover')
-      or (not client.resolved_capabilities.signature_help and method == 'textDocument/signatureHelp')
-      or (not client.resolved_capabilities.goto_definition and method == 'textDocument/definition')
-      or (not client.resolved_capabilities.implementation and method == 'textDocument/implementation')
-      or (not client.resolved_capabilities.declaration and method == 'textDocument/declaration')
-      or (not client.resolved_capabilities.type_definition and method == 'textDocument/typeDefinition')
-      or (not client.resolved_capabilities.document_symbol and method == 'textDocument/documentSymbol')
-      or (not client.resolved_capabilities.workspace_symbol and method == 'textDocument/workspaceSymbol')
-    then
+    local required_capability = req_to_capability[method]
+    if required_capability and not client.resolve_capabilities[required_capability] then
```

Comment:
I was afraid this could be breaking changing, as a method could not have a mapped capability (for example, textDocument/formatting isn't currently mapped). So in that case I decided to keep the call if required_capability is `nil`, which preserves the previous behavior. Alternatively I can make the mapping exhaustive, what do you think?